wp-includes/js/service-worker.js

```diff
@@ -0,0 +1,148 @@
+/* global workbox */
+
+/**
+ * Handle registering caching strategies.
+ */
+
+if ( ! self.wp ) {
+	self.wp = {};
+}
+
+wp.serviceWorker = workbox;
+
+/**
+ * Custom router for handling conflicts between registered routes.
+ */
+class WPRouter extends wp.serviceWorker.routing.Router {
+
+	_findHandlerAndParams( event, url ) {
+		const routes = this._routes.get( event.request.method ) || [];
+		let matches = 0,
+			matchResult,
+			matchedRoute = null;
+		for ( const route of routes ) {
+			matchResult = route.match( { url, event } );
+			if ( matchResult ) {
+				matches++;
+				matchedRoute = route;
+			}
+		}
+
+		// If we didn't have a match, then return undefined values.
+		if ( 0 === matches ) {
+			return { handler: undefined, params: undefined };
+		} else if ( 1 === matches ) {
+			// If there was exactly one match.
+			if ( Array.isArray( matchResult ) && 0 === matchResult.length ) {
+				// Instead of passing an empty array in as params, use undefined.
+				matchResult = undefined;
+			} else if ( matchResult.constructor === Object && Object.keys( matchResult ).length === 0 || matchResult === true ) {
+				// Instead of passing an empty object in as params, use undefined.
+				matchResult = undefined;
+			}
+
+			return {
+				matchedRoute,
+				params: matchResult,
+				handler: matchedRoute.handler
+			};
+
+			// @todo Confirm approach for conflicting routes: for example perhaps we shouldn't consider multiple matches but same strategy a conflict. Maybe we should still use the first matching route and warn about others.
```

Comment:
Currently this approach is skipping all the routes where more than one match was found and logging the warning like this:
![screen shot 2018-08-06 at 2 41 11 pm](https://user-images.githubusercontent.com/3294597/43715474-46f4e676-9989-11e8-9330-0e6e5a5c9387.png)

Thinking that instead of this approach maybe we should take the first match as it is in the default behavior and report the conflicts. However, perhaps if the strategy is the same (e.g. two matching registered routes with the same strategy) then this is actually not a conflict and not report this. So basically report all the matching routes where the strategy differs. Just realized that there could still be conflicts when the expiration time configuration differs though. Thoughts, @westonruter?

Reply:
Just realized I commented on this todo in another PR: https://github.com/xwp/pwa-wp/pull/48#discussion_r208068200

>  I think it's a good idea to let the first match handle the request, and then to warn about the conflicting handlers.

Your right that we need to report any time that more than one match occurs, since different caches could be used or different plugins employed.